### PR TITLE
Preliminary ELG cuts for DR8 imaging for main and SV

### PR DESCRIPTION
**Pull request by geordie666**
This PR is a preliminary update to the ELG selection in preparation for DR8.

It may be the case that we can deprecate the 4 "sub-bits" for the ELG SV selection, but I'm retaining those bit-slots in the SV yaml files, for now, in an attempt to not break downstream code. If the final ELG SV selection only demands one bit, then I'll update the SV yaml file at that point.

### CODE REVIEW

**Comment by geordie666** [comment]
@moustakas: A heads-up...when this is merged it will almost certainly break the mocks, as I'm now passing `MASKBITS` as an extra parameter in `cuts._prepare_optical_wise()`.

**Comment by geordie666** [comment]
I've asked Anand for a review offline. I'll merge this once he confirms that I have the cuts correct.

**Comment by geordie666** [comment]
@moustakas: I'm about to merge this, which, I think, will break the mocks.

At some point soon I'm going to need to make a tag with which to run DR8 targeting, and presumably @sbailey would like it if tagged versions didn't break his integrated unit tests. So, it would be excellent if you could make the few fixes required for the mocks relatively soon. 

Otherwise, this is a heads-up to @sbailey that he might have to bypass the mocks in his unit tests, which I think he's done before.

File: py/desitarget/cuts.py [hunk 1]
```diff
@@ -255,91 +256,72 @@ def isLRGpass(gflux=None, rflux=None, zflux=None, w1flux=None, w2flux=None,
...
 
-    elg &= notinELG_mask(gallmask=gallmask, rallmask=rallmask, zallmask=zallmask,
-                         brightstarinblob=brightstarinblob, primary=primary)
+    elg &= notinELG_mask(maskbits=maskbits, gsnr=gsnr, rsnr=rsnr, zsnr=zsnr,
+                         primary=primary)
 
-    elg &= isELG_colors(gflux=gflux, rflux=rflux, zflux=zflux, w1flux=w1flux, w2flux=w2flux,
-                        south=south, primary=primary)
+    elg &= isELG_colors(gflux=gflux, rflux=rflux, zflux=zflux, south=south,
+                        primary=primary)
```
Comment (moustakas) [inline py/desitarget/cuts.py]: Yeah, these changes broke the mocks.  Even though `w1flux` and `w2flux` aren't used for ELG target selection, the way the targeting cuts are applied to the mocks is to pass all the *grz* photometry independent of the class of object.

Comment (geordie666) [reply py/desitarget/cuts.py]: Oops, my bad. I should have remembered *that*. I'll restore this branch and fix that in the next 20 minutes.

Comment (geordie666) [reply py/desitarget/cuts.py]: Actually, I'll just fix this in a separate PR for better transparency. I'll ask for your review in the next PR.

Comment (moustakas) [reply py/desitarget/cuts.py]: I have the changes in a branch already and I'm testing other permutations of the code. I can open a simple PR (or even push the very minor change to master...).  Let me know.

Comment (geordie666) [reply py/desitarget/cuts.py]: Will fix in #514. Thanks.

**Pull request merged**